aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -190,8 +192,7 @@ module aptos_framework::object {
 
     /// Create a new object whose address is derived based on the creator account address and another object.
     /// Derivde objects, similar to named objects, cannot be deleted.
-    public fun create_user_derived_object(creator: &signer, derive_ref: &DeriveRef): ConstructorRef {
-        let creator_address = signer::address_of(creator);
+    public(friend) fun create_user_derived_object(creator_address: address, derive_ref: &DeriveRef): ConstructorRef {
```

Comment:
Why?

Reply:
you don't need a signer now.
Another rationale is offering users to use this derive_ref arbitrarily would potentially make difference resources using the same object. I am not sure this is something we want.
cc @davidiw @CapCap